### PR TITLE
fill in missing changelog entries based on pull request numbers in commit history

### PR DESCRIPTION
**Pull request by zacharyburnett**
some changes made in pull requests had not been documented in the change log. I used the following script to find discrepancies between the commit history and `CHANGES.rst`:
```python
import re
import sys
from pathlib import Path

from git import Repo

changelog_pull_request_pattern = re.compile('\[(#\d+)\]')
commit_pull_request_pattern = re.compile('#\d+')

repo = Repo('.')

commit_messages = '\n'.join([commit.message for commit in repo.iter_commits()])
commit_pull_requests = re.findall(commit_pull_request_pattern, commit_messages)

with open(Path(repo.working_dir) / 'CHANGES.rst') as change_log:
    changelog_pull_requests = re.findall(changelog_pull_request_pattern, change_log.read())

for pull_request in commit_pull_requests:
    if pull_request not in changelog_pull_requests:
        print(f'pull request {pull_request} not in changelog', file=sys.stderr)
```

Additionally, I copied the change log check workflow from `asdf` (https://github.com/spacetelescope/stcal/pull/100) so if that is merged then this should no longer be an issue

### CODE REVIEW

**Comment by zacharyburnett** [comment]
the remaining pull requests seem to be trivial changes:
```
pull request #93 not in changelog
pull request #88 not in changelog
pull request #86 not in changelog
pull request #80 not in changelog
pull request #77 not in changelog
pull request #73 not in changelog
pull request #62 not in changelog
pull request #56 not in changelog
pull request #53 not in changelog
pull request #46 not in changelog
pull request #46 not in changelog
pull request #48 not in changelog
pull request #29 not in changelog
pull request #28 not in changelog
pull request #15 not in changelog
pull request #11 not in changelog
pull request #9 not in changelog
pull request #8 not in changelog
pull request #1 not in changelog
```

File: CHANGES.rst [hunk 1]
```diff
@@ -132,6 +137,11 @@ dark_current
 0.4.3 (2021-12-27)
 ==================
 
+general
+-------
+
+- Use new ``publish-to-pypi.yml`` to build & publish wheels [#69]
```
Comment (nden) [inline CHANGES.rst:143]: I would remove this one as not interesting to users

File: CHANGES.rst [hunk 2]
```diff
@@ -205,23 +225,35 @@ Workaround for setuptools_scm issues with recent versions of pip. [#45]
...
+- Fix ramp fitting multiprocessing. [#30]
 
 
 0.2.2 (2021-07-19)
 ==================
 
+general
+-------
+
+- Add pypi publish workflow and fix jwst CI [#21]
```
Comment (nden) [inline CHANGES.rst:244]: remove this one

**Review by nden — CHANGES_REQUESTED** [review]

**Comment by zacharyburnett** [comment]
do you think there should also be an `Other` section for changes to documentation or build configuration?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+<!-- describe the changes comprising this PR here -->
+This PR addresses ...
+
+**Checklist**
+- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
+- [ ] updated relevant tests
+- [ ] updated relevant documentation
+- [ ] updated relevant milestone(s)
+- [ ] added relevant label(s)
```
Comment (zacharyburnett) [inline .github/pull_request_template.md:16]: I copied the pull request template from `jwst` and modified it to explicitly ask to update `CHANGES.txt` in the relevant section

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<!-- If this PR closes a JIRA ticket, make sure the title starts with the JIRA issue number, 
+for example JP-1234: <Fix a bug> -->
+Resolves [JP-nnnn](https://jira.stsci.edu/browse/JP-nnnn)
+
```
[the remark this answers is not in the document]
Comment (zacharyburnett) [reply .github/pull_request_template.md:5]: done in 18a9ad1

**Comment by zacharyburnett** [comment]
Posting the script here so we can use it later if it is needed:
```python
import re
import sys
from pathlib import Path

from git import Repo

changelog_pull_request_pattern = re.compile('\[(#\d+)\]')
commit_pull_request_pattern = re.compile('#\d+')

repository = Repo('.')
changelog_filename = Path(repository.working_dir) / 'CHANGES.rst'

commit_messages = '\n'.join([commit.message for commit in repository.iter_commits(rev=None)])
commit_pull_requests = re.findall(commit_pull_request_pattern, commit_messages)

print(f'found {len(commit_pull_requests)} pull requests in main branch')

with open(changelog_filename) as change_log:
    changelog_pull_requests = re.findall(changelog_pull_request_pattern, change_log.read())

print(f'found {len(changelog_pull_requests)} pull requests in "{changelog_filename.name}"')

missing_commit_pull_requests = []
for pull_request in commit_pull_requests:
    if pull_request not in changelog_pull_requests:
        missing_commit_pull_requests.append(pull_request)
if len(missing_commit_pull_requests) > 0:
    print(
        f'found {len(missing_commit_pull_requests)} pull requests from "{changelog_filename.name}" '
        f'that are not in commit history: \n'
        f'{" ".join(missing_commit_pull_requests)}',
        file=sys.stderr,
    )

missing_changelog_pull_requests = []
for pull_request in changelog_pull_requests:
    if pull_request not in commit_pull_requests:
        missing_changelog_pull_requests.append(pull_request)
if len(missing_changelog_pull_requests) > 0:
    print(
        f'found {len(missing_changelog_pull_requests)} pull requests from commit history '
        f'that are not in "{changelog_filename.name}": \n'
        f'{" ".join(missing_changelog_pull_requests)}',
        file=sys.stderr,
    )
```

**Pull request merged**